### ISSUE TITLE
Applying 'IDE0004: Cast is redundant' for '(UnityEngine.Object)null' can break things

### ISSUE THREAD

**Issue by aybe** [microsoft/Microsoft.Unity.Analyzers#117]
#### Problem statement

For the story, I am implementing material extraction for a custom `ScriptedImporter`, reusing the code from `ModelImporterMaterialEditor` internal class in Unity.

Take a look at this:

![devenv_2020-11-15_18-21-27](https://user-images.githubusercontent.com/1537591/99191800-68eabf00-276f-11eb-881b-0a13b05b0dd5.png)

If you apply the fix then, for instance in this case, undo/redo will stop working.

Basically, the cast to `(Object)` of `null` is an absolute necessity.

#### Proposed solution

Maybe a warning of some sort to tell about potential issue in doing so?

**Comment by aybe** [comment]
Sorry, false flag actually 🏳😂
